src/rust/engine/src/nodes.rs

```diff
@@ -767,12 +787,19 @@ pub struct ProcessResult(process_executor::ExecuteProcessResult);
 impl Node for ExecuteProcess {
   type Output = ProcessResult;
 
-  fn run(self, _: Context) -> NodeFuture<ProcessResult> {
+  fn run(self, context: Context) -> NodeFuture<ProcessResult> {
     let request = self.0.clone();
 
-    let tmpdir = TempDir::new("process-execution").unwrap();
-    // TODO: Materialize the snapshot into the tmpdir when ExecuteProcessRequest has a Snapshot.
+    // TODO: Make this much less unwrap-happy as part of moving this to run off-thread and
+    // asynchronously.
```

Comment:
ditto snapshot manipulation reference

Reply:
Filed https://github.com/pantsbuild/pants/issues/5502 and linked to it there